File: MCDA-Database_project/finalsubmission/scripts/Article_Author.py
import pandas as pd
import ast

author = pd.read_csv(r"authors.csv")
author_id = author['_id'].tolist()
author_name = list(author['fname'].values + ' ' + author['lname'].values)
article = pd.read_csv('articles.csv')
article_id = article['article_id'].tolist()
article_author = article['authors'].tolist()
article_author = [ast.literal_eval(i) for i in article_author]
result = []
count = 0
for i in range(len(article_author)):
    for j in range(len(article_author[i])):
        result.append([count, author_id[author_name.index(article_author[i][j])], article_id[i]])
        count = count + 1
pd.DataFrame(result).to_csv("article_author.csv", index=False, header=['art_auth_id', 'author_id', 'article_id'])

File: MCDA-Database_project/finalsubmission/scripts/article_author.js
var auto_id = 0;
var ptr = db.ARTICLE.find();
var ptr1= db.AUTHOR.find();

while (ptr.hasNext()){

    var doc = ptr.next();
       
        if(Array.isArray(doc.author))
        {
         for (var i=0;i<doc.author.length;i++)
               authors[i]=doc.author[i].ftext;
                 while (ptr1.hasNext())
               {

                 var doc1=ptr1.next();
                 str author =doc1.fname +""+doc1.lname;
                 if (str == authors[i])
                {
                     var row = {"art_auth_id":auto_id,"author_id":doc1._id, "article_id": doc.article_id};
                 db.ARTICLE_AUTHOR.insert(row);
                 auto_id++;
               }
         }
         else
            authors[0]=doc.author.ftext; 
             while (ptr1.hasNext())
               {

                 var doc1=ptr1.next();
                 str author =doc1.fname +""+doc1.lname;
                 if (str == authors[0])
                     var row = {"art_auth_id":auto_id,"author_id":doc1._id, "article_id": doc.article_id};
                 db.ARTICLE_AUTHOR.insert(row);
                 auto_id++;
               }

}
    /* var authors= new Array();   
         
        // for(i in doc.author)
        // authors.push(doc.author[i].ftext)
        
         if(Array.isArray(doc.author))
        {
         for (var i=0;i<doc.author.length;i++)
               authors[i]=doc.author[i].ftext;
         }
         else
            authors[0]=doc.author.ftext;
  
        var row = {"article_id":auto_id,"article_title":doc.title.ftext,"name":doc.journal.ftext, "vol_id":doc.volume.ftext,
           "year":doc.year.ftext, "page_num":doc.pages.ftext,"authors":authors};
        db.ARTICLE.insert(row);
        auto_id++;
    */



